src/helpers/connector.ts

```diff
@@ -4,18 +4,41 @@ import {
   StarknetkitConnector,
 } from "../connectors"
 
+export function isCompoundConnector(
+  connector: Connector | StarknetkitConnector | StarknetkitCompoundConnector,
+): boolean {
+  return (connector as StarknetkitCompoundConnector).isCompoundConnector
+}
+
 export function extractConnector(
   connector: Connector | StarknetkitConnector | StarknetkitCompoundConnector,
   useFallback: boolean = false,
 ) {
-  if ((connector as StarknetkitCompoundConnector).isCompoundConnector) {
+  const isCompound = isCompoundConnector(connector)
+
+  if (isCompound) {
     return useFallback
       ? (connector as StarknetkitCompoundConnector).fallbackConnector
       : (connector as StarknetkitCompoundConnector).connector
   }
   return connector as StarknetkitConnector
 }
 
+export function getConnectorMeta(
```

Comment:
maybe `getConnectorNameAndIcon`?
just easier to understand than `Meta`

Reply:
I'd keep it meta, never know what could end up there. Name and icon are meta, plus this is internal function, not for users.